Modules/_io/fileio.c

```diff
@@ -699,6 +699,8 @@ _io_FileIO_readall_impl(fileio *self)
 
     if (self->fd < 0)
         return err_closed();
+    if (!self->readable)
+        return err_mode("reading");
```

Comment:
The [PEP 7](https://www.python.org/dev/peps/pep-0007/#code-lay-out) style guide states that "all new C code requires braces".